### PR TITLE
Add a fuzz target for SQL query

### PR DESCRIPTION
**Pull request by tomrittervg**
This adds a fuzz target for the Query() api.

It also includes the sql dictionary from AFL (which I think is far more featured than what osquery supports...) and has a pretty gnarly shell script to generate a seed corpus.  As described in the script, if it isn't perfect it's not a big deal because it's just to generate a starting point for the fuzzer. That said it produces pretty good results today, and it looks for pretty standard code constructs so I think (?) it will perform reasonably well into the future...

Part of #5905

### CODE REVIEW

**Comment by theopolis** [comment]
Do you mind separating out the corpus generation and dictionary content into a separate PR? I ask this because I'd like to gather more feedback about where to put these scripts and 'config data'.

File: osquery/main/harnesses/CMakeLists.txt [hunk 1]
```diff
@@ -5,6 +5,11 @@
 # the LICENSE file found in the root directory of this source tree.
 
 function(osqueryMainHarnesses)
+  if(OSQUERY_ENABLE_FUZZER_SANITIZERS AND NOT OSQUERY_BUILD_TESTS)
+    # Some fuzzers use include files from the tests
```
Comment (theopolis) [inline osquery/main/harnesses/CMakeLists.txt]: Interesting, this should not be the case.

Comment (tomrittervg) [reply osquery/main/harnesses/CMakeLists.txt]: It's because of sql_test_utils.h

**Comment by tomrittervg** [comment]
> Do you mind separating out the corpus generation and dictionary content into a separate PR? I ask this because I'd like to gather more feedback about where to put these scripts and 'config data'.

Sure, I'll try to get to it in a day or two.

**Comment by theopolis** [comment]
IMO we should create a directory within `./tools` (e.g., `./tools/harnesses`) and place the corpus-generation + dictionary content there.

File: osquery/main/harnesses/fuzz_sqlquery.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+#include <osquery/config/config.h>
+#include <osquery/core.h>
+#include <osquery/database.h>
+#include <osquery/logger.h>
+#include <osquery/registry.h>
+#include <osquery/sql.h>
+#include <osquery/sql/dynamic_table_row.h>
+#include <osquery/sql/tests/sql_test_utils.h>
```
Comment (theopolis) [inline osquery/main/harnesses/fuzz_sqlquery.cpp]: But, why do you need this include?

Comment (tomrittervg) [reply osquery/main/harnesses/fuzz_sqlquery.cpp]: Huh; I guess I don't. I swore I got an error when I omitted it in my initial attempt on a different branch; but now when I remove it, it works...

**Comment by tomrittervg** [comment]
> IMO we should create a directory within `./tools` (e.g., `./tools/harnesses`) and place the corpus-generation + dictionary content there.

Did you still want me to split them out; or is this a good place?  (Can always move them easily enough...)

File: tools/harnesses/gen_fuzz_sqlquery_corpus.sh [hunk 3]
```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
```
Comment (theopolis) [inline tools/harnesses/gen_fuzz_sqlquery_corpus.sh]: Mind replacing this with
```
#!/usr/bin/env bash

#  Copyright (c) 2014-present, Facebook, Inc.
#  All rights reserved.
#
#  This source code is licensed in accordance with the terms specified in
#  the LICENSE file found in the root directory of this source tree.
```

Comment (theopolis) [reply tools/harnesses/gen_fuzz_sqlquery_corpus.sh]: I suggest naming this `osqueryfuzz_sqlquery_corpus.sh` to conform with the harness binary+dict naming.

File: tools/harnesses/gen_fuzz_sqlquery_corpus.sh [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+# So we make a good effort to extract complete statements, but if it misses some, that's okay.
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+function usage() {
+    echo "${BASH_SOURCE[0]} destination-file"
+}
+
+function main() {
+    if [[ $# < 1 ]]; then
```
Comment (theopolis) [inline tools/harnesses/gen_fuzz_sqlquery_corpus.sh]: Also, we've been using `  ` (2 spaces) for indents in bash.

**Comment by tomrittervg** [comment]
Updated! LMK if I missed anything.

**Pull request merged**